src/benchmarklib/jcch/jcch_benchmark_item_runner.cpp

```diff
@@ -0,0 +1,324 @@
+          << _dbgen_path << "/dists.dss -r " << seed << " -l params >/dev/null";
+      auto ret = system(cmd.str().c_str());
+      Assert(!ret, "Calling qgen failed");
+    }
+
+    std::cout << " (" << timer.lap_formatted() << ")" << std::endl;
+  }
+
+  // Open the params file, which looks like this:
+  //   query_id|param0|param1
+  auto file = std::ifstream(params_path);
+  Assert(file.is_open(), std::string{"Could not open JCC-H parameters at "} + params_path);
+
+  std::string line;
+  while (std::getline(file, line)) {
+    // Load the parameter into the corresponding entry in _all_params
+    auto string_values = split_string_by_delimiter(line, '\t');
+    const auto query_id = std::stoi(string_values[0]);
+    Assert(query_id >= 1 && query_id <= 22, "Invalid query_id");
+    string_values.erase(string_values.begin());
+    _all_params[query_id - 1].emplace_back(string_values);
+  }
+}
+
+bool JCCHBenchmarkItemRunner::_on_execute_item(const BenchmarkItemID item_id, BenchmarkSQLExecutor& sql_executor) {
+  using namespace std::string_literals;  // NOLINT
+
+  const auto& this_item_params = _all_params[item_id];
+
+  // Choose a random parameterization from _all_params
+  static thread_local std::minstd_rand random_engine{_random_seed++};
+  std::uniform_int_distribution<> params_dist{0, static_cast<int>(this_item_params.size() - 1)};
+  const auto raw_params_iter = this_item_params.begin() + params_dist(random_engine);
+
+  std::vector<std::string> parameters;
+  auto sql = std::string{};
+
+  // This mirrors TPCHBenchmarkItemRunner::_on_execute_item. Instead of generating random parameters according to the
+  // TPC-H specifications, it uses the ones generated by JCC-H's qgen
+  switch (item_id) {
+    // Writing `1-1` to make people aware that this is zero-indexed while TPC-H query names are not
+    case 1 - 1: {
+      // In some cases, we still need to do the date calculations that Hyrise does not support yet
+      const auto date =
+          _calculate_date(boost::gregorian::date{1998, 12, 01}, 0, -std::atoi(raw_params_iter->at(0).c_str()));
+      parameters.emplace_back("'"s + date + "'");
+      break;
+    }
+
+    case 2 - 1: {
+      parameters.emplace_back(raw_params_iter->at(0));
+      parameters.emplace_back("'%"s + raw_params_iter->at(1) + "'");
+      parameters.emplace_back("'"s + raw_params_iter->at(2) + "'");
+      parameters.emplace_back("'"s + raw_params_iter->at(2) + "'");
+      break;
+    }
+
+    case 3 - 1: {
+      parameters.emplace_back("'"s + raw_params_iter->at(0) + "'");
+      parameters.emplace_back("'"s + raw_params_iter->at(1) + "'");
+      parameters.emplace_back("'"s + raw_params_iter->at(1) + "'");
+      break;
+    }
+
+    case 4 - 1: {
+      const auto begin_date = boost::gregorian::from_string(raw_params_iter->at(0));
+      const auto end_date = _calculate_date(begin_date, 3);
+
+      parameters.emplace_back("'"s + raw_params_iter->at(0) + "'");
```

Comment:
why not using begin_date here?

Reply:
Renamed `end_date` to `end_date_str` and added:

```
      // Cannot use begin_date here, as we would have to convert it into a string first.
      parameters.emplace_back("'"s + raw_params_iter->at(0) + "'");
```